### PR TITLE
Add list of dicts support for astra writing and reading

### PR DESCRIPTION
**Pull request by henrikfoss**

### CODE REVIEW

**Comment by henrikfoss** [comment]
create_package

**Comment by henrikfoss** [comment]
create_package

**Comment by henrikfoss** [comment]
create_package

**Comment by henrikfoss** [comment]
create_package

File: adapta/storage/distributed_object_store/datastax_astra/astra_client.py [hunk 1]
```diff
@@ -390,6 +390,7 @@ def map_to_column(  # pylint: disable=R0911
             typing.Tuple[Type[Column],],
             typing.Tuple[Type[Column], Type[Column]],
             typing.Tuple[Type[Column], Type[Column], Type[Column]],
+            typing.Tuple[Type[columns.List], Type[columns.Map]],
```
Comment (matt035343) [inline adapta/storage/distributed_object_store/datastax_astra/astra_client.py:393]:
```suggestion
            typing.Tuple[Type[columns.List], columns.Map],
```

File: adapta/storage/distributed_object_store/datastax_astra/astra_client.py [hunk 2]
```diff
@@ -414,6 +415,16 @@ def map_to_column(  # pylint: disable=R0911
...
                 return (columns.Text,)
             if get_origin(python_type) == list:
+                args = typing.get_args(python_type)
+                if get_origin(args[0]) == dict:
+                    dict_args = typing.get_args(args[0])
+                    return (
+                        columns.List,
+                        columns.Map(
+                            map_to_column(dict_args[0])[0],
+                            map_to_column(dict_args[1])[0],
```
Comment (matt035343) [inline adapta/storage/distributed_object_store/datastax_astra/astra_client.py:425]: Beware this assumes that the dict type can only have simple keys and values.

Structures like `list[dict[str,dict[str,str]]]` will not work here. Actually, same goes for the other list and map types in this function.

If you do not fix this, please open a new issue for this

**Review by matt035343 — APPROVED** [review]

File: adapta/storage/distributed_object_store/datastax_astra/astra_client.py (hunk 1 above)
Comment (henrikfoss) [reply adapta/storage/distributed_object_store/datastax_astra/astra_client.py]: fixed

File: adapta/storage/distributed_object_store/datastax_astra/astra_client.py (hunk 2 above)
Comment (henrikfoss) [reply adapta/storage/distributed_object_store/datastax_astra/astra_client.py:425]: Made this issue https://github.com/SneaksAndData/adapta/issues/425

**Pull request merged**